src/server.rs

```diff
@@ -374,12 +359,25 @@ fn handle_client(tcp_stream: Arc<Mutex<TcpStream>>, clients_set: &Arc<Mutex<Hash
             // FIXME: 7. Wait to receive the image files from all dependencies.
             // FIXME: 8. Send ACK to confirm that the image files from all checkpoints have been received.
         }
-    }
 
     // Close TCP connection with client
     client_close_connection(&client_id, client_action, tcp_stream, clients_set);
 }
 
+fn get_response_message(client_id: &str, clients_set: &Arc<Mutex<HashMap<String, ClientStatus>>>) -> &'static str {
+    let mut response_message = "ACK";
+    let mut clients_lock = clients_set.lock().unwrap();
+
+    if clients_lock.is_empty() || !clients_lock.contains_key(client_id) {
+        info!("[{}] [==] Insert client ID", client_id);
+        clients_lock.insert(client_id.to_string(), ClientStatus::new());
+    } else {
+        response_message = "client already connected";
```

Comment:
Would it be better to flip the condition in the if block and do an early return instead? Wondering if that makes the flow clearer.